src/gap.c

```diff
@@ -1416,15 +1414,16 @@ Obj FuncTIMEOUTS_SUPPORTED(Obj self) {
 Obj FuncCALL_WITH_TIMEOUT( Obj self, Obj seconds, Obj microseconds, Obj func, Obj args )
 {
   Obj res;
-  Obj currLVars;
+  volatile Obj currLVars;
   Obj result;
-  Int recursionDepth;
-  Stat currStat;
+  volatile Int recursionDepth;
+  volatile Stat currStat;
   UInt newJumpBuf = 1;
-  UInt mayNeedToRestore = 0;
-  Int iseconds, imicroseconds;
-  Int curr_seconds= 0, curr_microseconds=0, curr_nanoseconds=0;
+  volatile UInt mayNeedToRestore = 0;
+  volatile Int iseconds, imicroseconds;
+  volatile Int curr_seconds= 0, curr_microseconds=0, curr_nanoseconds=0;
   Int restore_seconds = 0, restore_microseconds = 0;
```

Comment:
Well, I only added the volatile to variables we access if `setjmp` returns a nonzero value.